### PR TITLE
Major perf improvement: support very wide Spark dataframe

### PR DESCRIPTION
**Pull request by andyndang**
* Improve performance when dealing with massive data frames
* Deprecate frequent number items (this is not used and store similar data to frequent items)

### CODE REVIEW

File: spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    b.merge(rhs)
+  }
+
+  override def merge(b1: DatasetProfile, b2: DatasetProfile): DatasetProfile =  {
+    b1.merge(b2)
+  }
+
+  override def finish(reduction: DatasetProfile): Array[Byte] = reduction.toBytes
+
+  override def bufferEncoder: Encoder[DatasetProfile] = Encoders.javaSerialization(classOf[DatasetProfile])
```
Comment (drewdahlke) [inline spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala:35]: Should we consider kyro?

File: spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                           sessionId: String = UUID.randomUUID().toString) extends Aggregator[Row, DatasetProfile, Array[Byte]] {
+  override def zero: DatasetProfile =  new DatasetProfile(sessionId, Instant.ofEpochMilli(sessionTimeInMillis))
+    .withTag("Name", datasetName)
+
+  override def reduce(b: DatasetProfile, row: Row): DatasetProfile = {
+    val is = new ByteArrayInputStream(row.getAs[Array[Byte]](PROFILE_FIELD));
+    val rhs = DatasetProfile.parse(is)
+    b.merge(rhs)
+  }
+
```
Comment (drewdahlke) [inline spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala:28]: Not sure if its necessary here, but most of the UDAF's where I've done profile merging end up with a pattern something like

    if (left == null || left.length == 0) {
      return right;
    }
    if (right == null || right.length == 0) {
      return left;
    }
// Then merge

Comment (andyndang) [reply spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala:28]: Yeah good call :D. I'll update the code

File: spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala (hunk 1 above)
Comment (andyndang) [reply spark/src/main/scala/org/apache/spark/whylogs/DatasetProfileMerger.scala:35]: No advantage. DatasetProfile implements protobuf-based serialization for Java serialization already :)

File: core/src/main/java/com/whylogs/core/SchemaInformation.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
 @UtilityClass
 public class SchemaInformation {
   final int SCHEMA_MAJOR_VERSION = 1;
-  final int SCHEMA_MINOR_VERSION = 2;
+  final int SCHEMA_MINOR_VERSION = 3;
```
Comment (drewdahlke) [inline core/src/main/java/com/whylogs/core/SchemaInformation.java:12]: Given a schema bump, its nice to have unit tests merging a newly generated profile against an older one in src/test/resources to make sure there aren't any merge issues against older profiles

Comment (andyndang) [reply core/src/main/java/com/whylogs/core/SchemaInformation.java:12]: I added a test for roundtrip parsing + merging - the parsing itself turns an old profile into a new profile in memory FYI

**Pull request merged**